### PR TITLE
Phase6 PyTest Parallel Execution

### PR DESCRIPTION
**Pull request by xenony**
Add pytest-xdist library
Add pytest-rerunfailures library
Add random test for check re-run

### CODE REVIEW

**Comment by xenony** [comment]
Report - https://67-178857803-gh.circle-artifacts.com/0/allure-report/index.html

File: tests/test_jira_api.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def test_login_to_jira(self, response, expected):
     @allure.title("Post issue")
     @pytest.mark.api
     def test_post_issue(self):
-        response = api.post_issue("OA API Issue001", "OlegAverkin", "Low")
+        response = api.post_issue("OA API Issue", "OlegAverkin", "Low")
         assert response.status_code == 201
```
Comment (archick12) [inline tests/test_jira_api.py:24]: Please, also check they type of content.
Also validate returned data, confirm that it contains required text.

**Review by archick12 — CHANGES_REQUESTED** [review]

File: tests/test_jira_api.py (hunk 1 above)
Comment (xenony) [reply tests/test_jira_api.py:24]: Done

**Review by archick12 — APPROVED** [review]
Good Job!

**Pull request merged**